src/arrayMethodReduce.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (!this.length) {
+      return initialValue;
+    }
+
+    let currentValue;
+    let startingIndex = 0;
+
+    if (arguments.length === 2) {
+      currentValue = initialValue;
+    } else {
+      currentValue = this[0];
+      startingIndex++;
+    }
```

Comment:
maybe just check if initialValue exists. If it does, assign it to a variable, otherwise assign this[0]. You can use ternary and do it in 1 line